src/main/scala/com/github/gvolpe/scalar2019/tagless/alt/laws.scala

```diff
@@ -0,0 +1,34 @@
+package com.github.gvolpe.scalar2019.tagless.alt
+
+import laws._
+
+object laws {
+  final case class IsEq[A](lhs: A, rhs: A)
+
+  implicit final class IsEqArrow[A](private val lhs: A) extends AnyVal {
+    def <->(rhs: A): IsEq[A] = IsEq(lhs, rhs)
+  }
+}
+
+trait MkDepLaws[F[- _, _], R] {
+  def M: MkDep[F, R]
+
+  /*
+   * Feeding R to F[R, A] eliminates R ang gives you F[Any, A] which is just F[A]
+   */
+  def elimination[A](fra: F[R, A], env: R, fa: F[Any, A]) = M[A](fra)(env) <-> fa
```

Comment:
How does a test construct an `F[R, A]` that, after elimination, is equal to the provided `F[Any, A]`? This may be a bit tautological...

Reply:
That is actually a good point. I thought about this in terms of `TaskR` but not in a fully polymorphic way. So my reasoning was:

- `elimination` basically represents `provide` for `TaskR / ZIO`.
- we can verify that the environment `R` is no longer present via `accessM`.

I'll be thinking about this and see if I can come up with a better proposal.